### DISCUSSION TITLE
All Users Randomly Lost App and Role Assignment

### DISCUSSION THREAD

**Discussion by crsnchng** [Budibase/budibase#5508] [Help] [answered]
Began working in my local Budibase install today and found that all users had lost their App and Role assignments. The app specific user table still reflected Role assignment but the global user list seemed to have been reset.

Unsure how one would be able to duplicate. I guess I just wanted to see if anyone else experienced this recently and if there is a way to prevent it.

Thanks in advance,

**Comment by crsnchng** [comment]
After some investigation - it seems to happen when an app goes from published to unpublished. 

Unsure if this is on purpose.

**Reply by melohagan** [reply]
It might be a bug. If you could reproduce the steps, it would be helpful to raise an issue. 
This bug might also be related? https://github.com/Budibase/budibase/issues/5155

**Reply by crsnchng** [reply]
@melohagan, I was able to replicate it. If I create users while the app is published, when I unpublish the app, all the roles are lost. When I publish the app again, roles do not come back.

**Reply by melohagan** [reply]
Thanks @fwit-carson I've raised a bug here: https://github.com/Budibase/budibase/issues/5522

**Comment by crsnchng** [comment, marked as answer]
Fixed in bug raised by @melohagan #5522
